File: backend/app/models/worker.py
from enum import Enum

from tortoise import models, fields


class WorkerStatus(str, Enum):
    ACTIVE = "aktiv"
    ON_LEAVE = "məzuniyyətdə"
    INACTIVE = "deaktiv"


class Worker(models.Model):
    id = fields.UUIDField(pk=True)
    fullname = fields.CharField(max_length=255)
    name = fields.CharField(max_length=255)
    surname = fields.CharField(max_length=255)
    email = fields.CharField(max_length=255)
    phone_number = fields.CharField(max_length=9)
    department = fields.TextField()
    started_work = fields.DateField()
    status = fields.CharEnumField(
        WorkerStatus, max_length=32, default=WorkerStatus.ACTIVE
    )

File: frontend/constant/worker.ts
import { StatusEnum } from "@/types/worker.types";

export const statusLabels: Record<StatusEnum, string> = {
  [StatusEnum.ACTIVE]: "Aktiv",
  [StatusEnum.ON_LEAVE]: "Məzuniyyətdə",
  [StatusEnum.INACTIVE]: "Deaktiv",
};

export const statusStyles: Record<StatusEnum, string> = {
  [StatusEnum.ACTIVE]: "bg-emerald-50 text-emerald-600",
  [StatusEnum.ON_LEAVE]: "bg-amber-50 text-amber-600",
  [StatusEnum.INACTIVE]: "bg-muted text-muted-foreground",
};
